TechJobs.java

```diff
@@ -0,0 +1,127 @@
+                } else {
+
+                    ArrayList<String> results = JobData.findAll(columnChoice);
+
+                    System.out.println("\n*** All " + columnChoices.get(columnChoice) + " Values ***");
+
+                    // Print list of skills, employers, etc
+                    for (String item : results) {
+                        System.out.println(item);
+                    }
+                }
+
+            } else { // choice is "search"
+
+                // How does the user want to search (e.g. by skill or employer)
+                String searchField = getUserSelection("Search by:", columnChoices);
+
+                // What is their search term?
+                System.out.println("\nSearch term: ");
+                String searchTerm = in.nextLine();
+                
+                if (searchField.equals("all")) {
+                    printJobs(JobData.findByValue(searchTerm));
+                } else {
+                    printJobs(JobData.findByColumnAndValue(searchField, searchTerm));
+                }
+            }
+        }
+    }
+
+    // ﻿Returns the key of the selected item from the choices Dictionary
+    private static String getUserSelection(String menuHeader, HashMap<String, String> choices) {
+
+        Integer choiceIdx;
+        Boolean validChoice = false;
+        String[] choiceKeys = new String[choices.size()];
+
+        // Put the choices in an ordered structure so we can
+        // associate an integer with each one
+        Integer i = 0;
+        for (String choiceKey : choices.keySet()) {
+            choiceKeys[i] = choiceKey;
+            i++;
+        }
+
+        do {
+
+            System.out.println("\n" + menuHeader);
+
+            // Print available choices
+            for (Integer j = 0; j < choiceKeys.length; j++) {
+                System.out.println("" + j + " - " + choices.get(choiceKeys[j]));
+            }
+
+            choiceIdx = in.nextInt();
+            in.nextLine();
+
+            // Validate user's input
+            if (choiceIdx < 0 || choiceIdx >= choiceKeys.length) {
+                System.out.println("Invalid choice. Try again.");
+            } else {
+                validChoice = true;
+            }
+
+        } while(!validChoice);
+
+        return choiceKeys[choiceIdx];
+    }
+
+
+
+    // Print a list of jobs
+    private static void printJobs(ArrayList<HashMap<String, String>> someJobs) {
+
+        int i = 0;
+        if (!someJobs.isEmpty()) {
+            while (i < someJobs.size()) {
+                System.out.printf("***** %n position type: %s %n name: %s%n employer: %s%n location: %s%n core competency: %s%n*****%n", someJobs.get(i).get("position type"), someJobs.get(i).get("name"), someJobs.get(i).get("employer"), someJobs.get(i).get("location"), someJobs.get(i).get("core competency"));
+                i++;
+            }
+        } else {
+            System.out.printf("We came back with no results.\n");

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.